Engine/csound_orc_compile.c

```diff
@@ -1935,7 +1935,8 @@ if (engineState != &csound->engineState) {
       continue;
     if (PARSER_DEBUG)
       csound->DebugMsg(csound, "Instr 0 check on opcode=%s\n", bp->t.opcod);
-    if (UNLIKELY((thread = oentry->thread) & 06 ||
+    thread =  oentry->init && oentry->perf ? 3 : (oentry->init ? 1 : 2);
+    if (UNLIKELY((thread & 06) ||
```

Comment:
This logic seems odd given that thread can only be 1,2, or 3. Not sure how either & 06 or !thread would ever be true.

Reply:
it does, but that was the original code.

Reply:
I had a look and I can't see that in the current code there's a situation where

```
!thread && bp->t.pftype != 'b'
```

either. The only "thread 0" opcodes are instr, endin etc.